src/common/templates/_dbv3.tpl

```diff
@@ -0,0 +1,559 @@
+    {{- end }}
+    {{- $dbType := .dbType }}
+    {{- if empty $dbType }}
+        {{- fail "ERROR: missing input argument - dbType" }}
+    {{- end }}
+    {{- $instanceName := .instanceName }}
+    {{- if empty $instanceName }}
+        {{- fail "ERROR: missing input argument - instanceName" }}
+    {{- end }}
+    {{- printf "%s_%s_%s" (upper $instanceName) (upper $dbType) (upper $name) }}
+{{- end }}
+
+{{/*
+Generate DB ESO Context Identifier 
+
+USAGE:
+{{- include "harnesscommon.dbv3.esoSecretCtxIdentifier" (dict "ctx" $ "dbType" "mongo" "scope" "" "instanceName" "") }}
+
+PARAMETERS:
+- ctx: Context
+- dbType: DB Type. Allowed values: mongo, timescaledb
+- scope: Scope of ESO Secret Context Identifier. Allowed Values: "local", "global"
+- instanceName: Name of the DB Instance. Required only when "scope" if "local"
+
+*/}}
+{{- define "harnesscommon.dbv3.esoSecretCtxIdentifier" }}
+    {{- $ := .ctx }}
+    {{- $dbTypeAllowedValues := dict "mongo" "" "timescaledb" "" }}
+    {{- $dbType := .dbType }}
+    {{- if not (hasKey $dbTypeAllowedValues $dbType) }}
+        {{- $errMsg := printf "ERROR: invalid value %s for input argument dbType" $dbType }}
+        {{- fail $errMsg }}
+    {{- end }}
+    {{- $scope := .scope -}}
+    {{- $esoSecretCtxIdentifier := "" }}
+    {{- if eq $scope "local" }}
+        {{- $instanceName := .instanceName }}
+        {{- if empty $instanceName }}
+            {{- fail "ERROR: missing input argument - instanceName" }}
+        {{- end }}
+        {{- $instanceName =  lower $instanceName }}
+        {{- $esoSecretCtxIdentifier = (include "harnesscommon.secrets.localESOSecretCtxIdentifier" (dict "ctx" $ "additionalCtxIdentifier" (printf "%s-%s" $instanceName $dbType) )) }}
+    {{- else if eq $scope "global" }}
+        {{- $esoSecretCtxIdentifier = (include "harnesscommon.secrets.globalESOSecretCtxIdentifier" (dict "ctx" $ "ctxIdentifier" $dbType )) }}
+    {{- else }}
+        {{- $errMsg := printf "ERROR: invalid value %s for input argument scope" $scope }}
+        {{- fail $errMsg }}
+    {{- end }}
+    {{- printf "%s" $esoSecretCtxIdentifier }}
+{{- end }}
+
+{{/*
+Generate K8S Env Spec for MongoDB Environment Variables
+
+USAGE:
+{{- include "harnesscommon.dbv3.mongoEnv" (dict "ctx" $ "database" "harness") | indent 12 }}
+
+PARAMETERS:
+REQUIRED:
+1. ctx
+2. database
+
+OPTIONAL:
+1. userVariableName
+2. passwordVariableName
+
+*/}}
+{{- define "harnesscommon.dbv3.mongoEnv" }}
+    {{- $ := .ctx }}
+    {{- $dbType := "mongo" }}
+    {{- $database := .database }}
+    {{- if empty $database }}
+        {{- fail "ERROR: missing input argument - database" }}
+    {{- end }}
+    {{- $instanceName := include "harnesscommon.dbv3.generateInstanceName" (dict "database" $database) }}
+    {{- if empty $instanceName }}
+        {{- fail "ERROR: invalid instanceName value" }}
+    {{- end }}
+    {{- $localDBCtx := get $.Values.mongo $instanceName }}
```

Comment:
the local values structure should be `.Values.database.mongo` . The path should match with global